lib/components/base/ShareModal.vue

```diff
@@ -0,0 +1,219 @@
+<script setup>
+import {
+  Button,
+  Modal,
+  ClipboardCopyIcon,
+  Avatar,
+  LinkIcon,
+  SendIcon,
+  GlobeIcon,
+  TwitterIcon,
+  MastodonIcon,
+  RedditIcon,
+} from '@/components'
+import { computed, ref } from 'vue'
+
+const props = defineProps({
+  header: {
+    type: String,
+    default: 'Share',
+  },
+  shareTitle: {
+    type: String,
+    default: 'Modrinth',
+  },
+  shareText: {
+    type: String,
+    default: 'Share',
+  },
+  link: {
+    type: Boolean,
+    default: false,
+  },
+})
+
+const shareModal = ref(null)
+
+const qrCode = ref(null)
+const qrImage = ref(null)
+const content = ref(null)
+
+const show = async (passedContent) => {
+  content.value = passedContent
+  if (props.link) {
+    const url =
+      'https://api.qrserver.com/v1/create-qr-code/?size=150x150&data=' +
+      encodeURIComponent(passedContent)
+    fetch(url).then(async (response) => {
+      qrImage.value = await response.blob()
+      qrCode.value = response.url
+    })
+  }
+  shareModal.value.show()
+}
+
+const copyImage = async () => {
+  const item = new ClipboardItem({ 'image/png': qrImage.value })
+  await navigator.clipboard.write([item])
+}
+
+const copyText = async () => {
+  await navigator.clipboard.writeText(content.value)
+}
+
+const sendEmail = computed(
+  () =>
+    `mailto:user@test.com
+    ?subject=${encodeURIComponent(props.shareTitle)}
+    &body=` + encodeURIComponent(`${props.shareText}\n${content.value}`)
+)
+
+const sendTweet = computed(
+  () =>
+    `https://twitter.com/intent/tweet?text=` +
+    encodeURIComponent(`${props.shareText}\n${content.value}`)
+)
+
+const sendToot = computed(
+  () =>
+    `https://mastodon.social/share?text=` +
```

Comment:
mastodon.social isn't the only Mastodon instance. Needs to have a prompt to enter the instance to use.